Sources/AWSSDKSwiftCore/Message/AWSRequest.swift

```diff
@@ -93,13 +93,36 @@ public struct AWSRequest {
             headers["Content-Type"] = "application/octet-stream"
         }
 
-        var head = HTTPRequestHead(version: HTTPVersion(major: 1, minor: 1), method: HTTPMethod.RAW(value: awsRequest.httpMethod), uri: awsRequest.url.absoluteString)
+        var head = HTTPRequestHead(
+          version: HTTPVersion(major: 1, minor: 1),
+          method: nioHTTPMethod(from: awsRequest.httpMethod),
+          uri: awsRequest.url.absoluteString
+        )
         let generatedHeaders = headers.map { ($0, $1) }
         head.headers = HTTPHeaders(generatedHeaders)
 
         return Request(head: head, body: try awsRequest.body.asData() ?? Data())
     }
 
+    fileprivate func nioHTTPMethod(from: String) -> HTTPMethod {
```

Comment:
Maybe add a note here

```
// We need to convert from the NIO type to a string when we sign the headers and attach the auth cookies.
```

since I took a bit to remember why the heck we were doing this anyway (and not just using NIO's builtin values instead of the stringly conversion.